install_config/certificate_customization.adoc

```diff
@@ -19,82 +19,277 @@ This can be done during an
+The Subject Alternative Names (SAN) should contain the following.
+
+* Two service nostnames:
++
+----
+docker-registry.default.svc.cluster.local
+docker-registry.default.svc            
+----
+
+* service IP address. Use the following command to get the Docker registry service IP address: 
++
+----
+oc get service docker-registry --template='{{.spec.clusterIP}}'
+----
+
+* public hostname. Use the following command to get the Docker registry public hostname:
++
+----  
+oc get route docker-registry --template '{{.spec.host}}'
+----
+
+For example, the server certificate should contain SAN details similar to the following: 
+
+----
+X509v3 Subject Alternative Name: 
+               DNS:docker-registry-public.openshift.com, DNS:docker-registry.default.svc, DNS:docker-registry.default.svc.cluster.local, DNS:172.30.2.98, IP Address:172.30.2.98
+----
+
+[[configuring-custom-certificates-wildcard]]
+=== Using a Wildcard Certificate with the Default Router
+
+You can configure the {product-title} default router with a default wildcard certificate. A default wildcard certificate provides a convenient way for 
+applications deployed in {product-title} to use out-of-the-box encryption without needing custom certificates. 
+
+[NOTE]
+====
+Default wildcard certificates are recommended for non-production environments only.
+====
+
+In order to configure a default wildcard certificate, a certificate must be provisioned that is valid for `*.<app_domain>`, where `<app_domain>` is the value of `openshift_master_default_subdomain` in the xref:../install_config/install/advanced_install.adoc#configuring-ansible[Ansible inventory file], by default `/etc/ansible/hosts`. Once provisioned, you will need to place your certificate, key and ca cert files on your Ansible host, and add the following line to your Ansible inventory.
+
+----
+openshift_hosted_router_certificate={"certfile": "/path/to/apps.c1-ocp.myorg.com.crt", "keyfile": "/path/to/apps.c1-ocp.myorg.com.key", "cafile": "/path/to/apps.c1-ocp.myorg.com.ca.crt"}
+----
+
+For example:
+
+----
+openshift_hosted_router_certificate={"certfile": "/home/cloud-user/star-apps.148.251.233.173.nip.io.cert.pem", "keyfile": "/home/cloud-user/star-apps.148.251.233.173.nip.io.key.pem", "cafile": "/home/cloud-user/ca-chain.cert.pem"}
+----
+
+Where the parameter values are: 
+
+* *certfile* contains the {product-title} router certificate.
+
+* *keyfile* contains the {product-title} wildcard key
+
+* *cafile* contains the root CA for this key and certificate. If an intermediate CA is in use it should contain both the intermediate and root CA.
+
+If these files are new to your {product-title} cluster, run an Ansible playbook to add these files to the {product-title} configuration files. 
+The playbook adds the certificate files to the *_/etc/origin/master/named_certificates/_* directory.
+
+----
+ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/openshift-cluster/config.yml
```

Comment:
Not sure this is a supported usage for ocp, we never use installation playbook to add the certificate files

Reply:
This came from @ikke-t issue https://github.com/openshift/openshift-docs/issues/7495. Please advise what is the correct playbook. redeploy-certificates.yml?
"Now as there are new files introduced to OpenShift you need to run ansible configure to make it configure the OpenShift configuration files. The cert files will end up into master's /etc/origin/master/named_certificates/ directory.
ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/openshift-cluster/config.yml"